### PR TITLE
Add ibmc driver

### PR DESCRIPTION
**Pull request by IamFive**
enable ibmc hardware type

### CODE REVIEW

**Comment by maelk** [comment]
looks good to me

**Comment by maelk** [comment]
quay.io is down, so no tests are passing today

**Comment by zaneb** [comment]
Given that the iBMC driver is [deprecated in Ironic](https://review.opendev.org/703465), I'm not sure that this makes sense for us?

**Comment by juliakreger** [comment]
@zaneb See: https://review.opendev.org/#/c/723225/ for it being un-deprecated.

**Comment by zaneb** [comment]
@juliakreger ah, thanks. That commit only reverts 1/4 of the `supported = False` things and, crucially, fails to remove the deprecation notice from the [docs](https://docs.openstack.org/ironic/latest/admin/drivers/ibmc.html). Here is a fix: https://review.opendev.org/729355

**Comment by juliakreger** [comment]
@zaneb Doh! Thanks!

**Comment by IamFive** [comment]
@maelk I got a question. Will `driver-requirements.txt` auto installed? Should we update docker file, and install vendor driver dependent python libs?

**Comment by maelk** [comment]
I am not sure which file you are mentioning, do you have a pointer ? For now we install ironic through a package manager, dnf, so the dependencies should be taken care of, afaik

**Comment by IamFive** [comment]
The driver-requirements.txt i mentioned is : https://github.com/openstack/ironic/blob/master/driver-requirements.txt .

The vendor dependencies is not auto installed in the docker image. So, I update the docker file, https://github.com/metal3-io/ironic-image/pull/159/files#diff-3254677a7917c6c01f55212f86c57fbfR43-R45.

File: Dockerfile [hunk 1]
```diff
@@ -40,6 +40,9 @@ RUN dnf install -y python3 python3-requests && \
     dnf clean all && \
     rm -rf /var/cache/{yum,dnf}/*
 
+# install vendor driver libs
+RUN /usr/bin/pip3 install python-ibmcclient>=0.2.3
```
Comment (elfosardo) [inline Dockerfile:44]: please we should not install anything using pip but only rely on packages as for other requirements

Comment (IamFive) [reply Dockerfile]: hi @elfosardo , `python-ibmcclient` is available now at: https://trunk.rdoproject.org/centos8/component/baremetal/45/eb/45ebd10ea96a436f0ba6bd06dd393df26d333606_168399e5/

**Comment by IamFive** [comment]
hi @maelk , python-ibmcclient lib has been added to rdopackages, may we continue integration steps now.

File: Dockerfile [hunk 2]
```diff
@@ -40,6 +40,9 @@ RUN dnf install -y python3 python3-requests && \
     dnf clean all && \
     rm -rf /var/cache/{yum,dnf}/*
 
+# install vendor driver libs
+RUN dnf install python3-ibmcclient
+
```
Comment (maelk) [inline Dockerfile:45]: please include this as part of the previous RUN command. It is good practice to limit the number of RUN and in addition you would benefit from the cache cleanup.

**Comment by maelk** [comment]
In addition, Travis failed with :
```
No match for argument: python3-ibmcclient

Error: Unable to find a match: python3-ibmcclient

The command '/bin/sh -c dnf install python3-ibmcclient' returned a non-zero code: 1

Makefile:8: recipe for target 'docker' failed

make: *** [docker] Error 1

The command "make $TARGET" exited with 2.
```

**Comment by IamFive** [comment]
hi @maelk , I am not sure how rdopackage repo is added. `python-ibmcclient` lib was published yestoday(2020-06-01), you can find it here: https://trunk.rdoproject.org/centos8/component/baremetal/45/eb/45ebd10ea96a436f0ba6bd06dd393df26d333606_168399e5/.

To include new added libs, this repo(http://trunk.rdoproject.org/centos8-master/current/delorean.repo) should be replaced under /etc/yum.repos.d.

Hi, @dtantsur , may u help give a comment about this issue.

**Comment by IamFive** [comment]
sorry, my bad, did not see the `dnf clean all` thing. It should be ok now.

**Comment by elfosardo** [comment]
@IamFive we switched from current to current-tripleo repositories, that's why the package can't be found
The package needs to be promoted to current-tripleo before being accessible by the image

**Comment by IamFive** [comment]
hi all, python3-ibmcclient is available on [current-tripleo](https://trunk.rdoproject.org/centos8-master/component/baremetal/current-tripleo/) now.  May we start merge this PR.

**Comment by IamFive** [comment]
@zaneb  also need an approved

**Comment by zaneb** [comment]
> @zaneb also need an approved

I *know* I'm not cool enough for that because there's a [list](https://github.com/metal3-io/ironic-image/blob/master/OWNERS). Also AIUI the whole idea is to get two different people.

**Pull request merged**